src/components/productdetail/VariantSelector.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <div v-if="product"
+       class="row select-row">
+    <ul class="list-inline"
+        data-test="variant-selector-list">
+      <li v-for="(values, name) in attributes"
+          :key="name">
+        <p class="text-uppercase"
+           data-test="attribute-name">
+          {{name}}
+        </p>
+        <AttributeSelect :product="product"
+                         :values="values"
+                         :sku="sku"
+                         :name="name"
+                         @updateCombi="updateSku"
+                         :selected="selected"
+                         :variantCombinations="variantCombinations" />
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+import flatMap from 'lodash.flatmap';
+import AttributeSelect from './AttributeSelect.vue';
+
+export default {
+  components: { AttributeSelect },
+
+  props: {
+    sku: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data: () => ({
+    product: null,
+    selected: {},
+  }),
+
+  watch: {
+    variantCombinations(value) {
```

Comment:
Instead of being in a watch, this could be a computed property.
Here some info about computed vs watch: https://vuejs.org/v2/guide/computed.html#Computed-vs-Watched-Property

Reply:
I tried to write it as a computed property but the problem is that when it's computed I have to call/render it somewhere which I don't want to do because the purpose of this is to just display the current variants in the dropdown on loading the page.
So to have it in watch makes more sense for me, what do you think?

Reply:
Hmmm... this property is passed as a prop to the child component. That should be enough to enable it as an observable. So you mean this is not working as this? Let's check it together.